src/Microsoft.OData.Core/UriParser/Parsers/FunctionOverloadResolver.cs

```diff
@@ -139,19 +139,20 @@ internal static bool ResolveOperationFromList(string identifier, IEnumerable<str
                 }
             }
 
-            IEnumerable<IEdmOperation> candidateMatchingOperations = null;
+            IEnumerable<IEdmOperation> operationsFromModel = null;
 
             // The extension method FindBoundOperations & FindOperations call IEdmModel.FindDeclaredBoundOperations which can be implemented by anyone and it could throw any type of exception
             // so catching all of them and simply putting it in the inner exception.
+            // It is better to cast the results of ResolveBoundOperations as IList instead of looping over this for basic things such as count.

```

Comment:
is this comment outdated? You don't cast here.

Reply:
Yes. I improved the logic to create a list while filtering instead of doing a ToList here. 

---
In reply to: [370842929](https://github.com/OData/odata.net/pull/1602#discussion_r370842929) [](ancestors = 370842929)